File: Difficulty: Easy/Dominant Pairs/dominant-pairs.cpp
class Solution {
  public:
    int dominantPairs(int n, vector<int> &arr) {
        int mid = n / 2;
        
        vector<int> left(arr.begin(), arr.begin() + mid);
        vector<int> right(arr.begin() + mid, arr.end());
        
        sort(left.begin(), left.end());
        sort(right.begin(), right.end());
        
        int count = 0;
        int j = 0;
        
        for (int i = 0; i < left.size(); i++) {
            while (j < right.size() && left[i] >= 5LL * right[j]) {
                j++;
            }
            count += j;
        }
        
        return count;
    }
};

File: Difficulty: Easy/Dominant Pairs/dominant-pairs.py
class Solution:
    def dominantPairs(self, arr: list[int]) -> int:
        # code here
        middle = len(arr) // 2
        l = sorted(arr[:middle])
        r = sorted(arr[middle:])
        
        win = 0 
        j = 0
        for a in l:
            while j < len(r) and a >= 5 * r[j]:
                j += 1
            win += j
            
        return win
